scripts/zkai_bounded_stwo_query_policy_hook_gate.py

```
@@ -0,0 +1,742 @@
+
+
+def find_stwo_source_root() -> pathlib.Path:
+    cargo_home = pathlib.Path(os.environ.get("CARGO_HOME", pathlib.Path.home() / ".cargo"))
+    registry_src = cargo_home / "registry" / "src"
+    candidates = sorted(registry_src.glob(f"*/stwo-{STWO_VERSION}"))
+    for candidate in candidates:
+        if all((candidate / rel).is_file() for rel in STWO_FILES.values()):
+            return candidate
+    raise BoundedStwoQueryPolicyHookGateError(
+        f"Stwo {STWO_VERSION} source is not available under {registry_src}; run cargo fetch"
+    )
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)